azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -723,16 +723,20 @@ void closeMBeans() {
    */
   @VisibleForTesting
   void shutdown() {
-    logger.info("Shutting down the pod");
-    final int execId = this.flowRunner.getExecutionId();
-    while (!this.flowFuture.isDone()) {
-      // This should not happen immediately as submitFlowRunner is a blocking call.
-      try {
-        Thread.sleep(100);
-      } catch (final InterruptedException e) {
-        logger.error(String.format("The sleep while waiting for execution : %d to finish was interrupted",
-            execId));
+    logger.info("Shutting down the container");
+    int execId = -1;
```

Comment:
The value -1 is never utilized in the code. All the places where execId is used, are the places where there is a check for flowRunner != null. Hence this initialization of execId can be done inside the if condition. 
For the uploadLogFile, it can be uploadLogFile(this.flowRunner.getExecutionId())